std/haxe/signals/ArraySignal.hx

```diff
@@ -1,10 +1,10 @@
-package haxe.async;
+package haxe.signals;
 
 /**
 	Basic implementation of a `haxe.async.Signal`. Uses an array for storing
 	listeners for the signal.
 **/
-class ArraySignal<T> implements Signal<T> {
+class ArraySignal<T> implements SignalEmitter<T> {
```

Comment:
As @back2dos mentioned on Slack, this implementation can go wrong in many ways, but perhaps that is the topic of an upcoming PR?

Reply:
I'm focusing on the API for now. Will start with implementations once API is settled.

Reply:
I'm focusing on the API for now. Will start with implementations once API is settled.